FAIL: test_ext4: fsconfig system call failed: /dev/sr1: Can't open blockdev

Version: 2.10.1 / HEAD
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1070019


Running the upstream test suite (via autopkgtest) triggers this failure in Debian sid:
```
======================================================================
FAIL: test_ext4 (__main__.FS.test_ext4)
fs: ext4
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest.tgUSRL/build.qT8/src/src/tests/integration-test", line 1107, in _do_udisks_check
    cd_fs.call_mount_sync(ro_options, None)
gi.repository.GLib.GError: udisks-error-quark: GDBus.Error:org.freedesktop.UDisks2.Error.Failed: Error mounting /dev/sr1 at /media/root/6e9fdd2f-90ee-4b54-b0d1-28b25e652896: fsconfig system call failed: /dev/sr1: Can't open blockdev (0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/autopkgtest.tgUSRL/build.qT8/src/src/tests/integration-test", line 725, in test_ext4
    self._do_fs_check('ext4')
  File "/tmp/autopkgtest.tgUSRL/build.qT8/src/src/tests/integration-test", line 894, in _do_fs_check
    self._do_udisks_check(fs_type)
  File "/tmp/autopkgtest.tgUSRL/build.qT8/src/src/tests/integration-test", line 1112, in _do_udisks_check
    self.fail('Mounting read-only device with \'rw\' option failed'
AssertionError: Mounting read-only device with 'rw' option failedwith an unexpected error.
Got: udisks-error-quark: GDBus.Error:org.freedesktop.UDisks2.Error.Failed: Error mounting /dev/sr1 at /media/root/6e9fdd2f-90ee-4b54-b0d1-28b25e652896: fsconfig system call failed: /dev/sr1: Can't open blockdev (0)
Expected: 'is write-protected but explicit read-write mode requested' or 'is write-protected but `rw' option given'

----------------------------------------------------------------------
Ran 31 tests in 292.422s

FAILED (failures=1)
```
This is with util-linux 2.40

Full test log is available at https://ci.debian.net/packages/u/udisks2/testing/amd64/46012968/

--- a/src/tests/integration-test
+++ b/src/tests/integration-test
@@ -1106,12 +1106,13 @@ class FS(UDisksTestCase):
                 ro_options = GLib.Variant('a{sv}', {'options': GLib.Variant('s', 'rw')})
                 cd_fs.call_mount_sync(ro_options, None)
             except GLib.GError as e:
-                new_msg = 'is write-protected but explicit read-write mode requested'
-                old_msg = 'is write-protected but `rw\' option given'
-                if not (old_msg in str(e) or new_msg  in str(e)):
-                    self.fail('Mounting read-only device with \'rw\' option failed'
+                msg = r"(is write-protected but `rw' option given|" \
+                        "is write-protected but explicit read-write mode requested|" \
+                        "fsconfig system call failed: /dev/sr.*: Can't open blockdev)"
+                if not re.search(msg, str(e)):
+                    self.fail('Mounting read-only device with \'rw\' option failed '
                               'with an unexpected error.\nGot: %s\nExpected: \'%s\''
-                              ' or \'%s\'' % (str(e), new_msg, old_msg))
+                              % (str(e), msg))
             else:
                 self.retry_busy(cd_fs.call_unmount_sync, no_options, None)
                 self.fail('Mounting read-only device didn\'t fail with \'rw\' option.')

